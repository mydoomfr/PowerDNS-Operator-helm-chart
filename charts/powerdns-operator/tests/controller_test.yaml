suite: test controller deployment
templates:
  - deployment.yaml
tests:
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
  - it: should set imagePullPolicy to Always
    set:
      image.pullPolicy: Always
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
  - it: should imagePullPolicy to be default value IfNotPresent
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: should override securityContext
    set:
      podSecurityContext:
        runAsUser: 2000
      securityContext:
        runAsUser: 3000
    asserts:
      - equal:
          path: spec.template.spec.securityContext
          value:
            runAsNonRoot: true
            runAsUser: 2000
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            runAsUser: 3000
            seccompProfile:
              type: RuntimeDefault
  - it: should override hostNetwork
    set:
      hostNetwork: true
    asserts:
      - equal:
          path: spec.template.spec.hostNetwork
          value: true
  - it: should override metrics port
    set:
      metrics.service.port: 8888
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[2]
          value: "--metrics-bind-address=:8888"
  - it: should override image version
    set:
      image.repository: ghcr.io/powerdns-operator/powerdns-operator
      image.tag: 0.2.0
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: ghcr.io/powerdns-operator/powerdns-operator:v0.2.0
  - it: should override resources
    set:
      resources:
        limits:
          cpu: 1000m
          memory: 256Mi
        requests:
          cpu: 20m
          memory: 128Mi
    asserts:
      - equal:
          path: spec.template.spec.containers[0].resources
          value:
            limits:
              cpu: 1000m
              memory: 256Mi
            requests:
              cpu: 20m
              memory: 128Mi
  - it: should override readinessProbe
    set:
      readinessProbe:
        httpGet:
          path: /fake-readyz
          port: 8082
        initialDelaySeconds: 10
        periodSeconds: 15
    asserts:
      - equal:
          path: spec.template.spec.containers[0].readinessProbe
          value:
            httpGet:
              path: /fake-readyz
              port: 8082
            initialDelaySeconds: 10
            periodSeconds: 15
  - it: should override livenessProbe
    set:
      livenessProbe:
        httpGet:
          path: /fake-readyz
          port: 8082
        initialDelaySeconds: 10
        periodSeconds: 15
    asserts:
      - equal:
          path: spec.template.spec.containers[0].livenessProbe
          value:
            httpGet:
              path: /fake-readyz
              port: 8082
            initialDelaySeconds: 10
            periodSeconds: 15
  - it: should override credentials (existing-secret)
    set:
      credentials:
        existingSecret: "existing-secret"
    asserts:
      - equal:
          path: spec.template.spec.containers[0].envFrom[0].secretRef.name
          value: existing-secret
  - it: should override credentials (new-secret)
    set:
      credentials:
        existingSecret: ""
        name: new-secret
        data:
          key1: value1
    asserts:
      - equal:
          path: spec.template.spec.containers[0].envFrom[0].secretRef.name
          value: new-secret
  - it: should override serviceAccountName
    set:
      serviceAccount:
        name: new-service-account
    asserts:
      - equal:
          path: spec.template.spec.serviceAccountName
          value: new-service-account
  - it: should override terminationGracePeriodSeconds
    set:
      terminationGracePeriodSeconds: 20
    asserts:
      - equal:
          path: spec.template.spec.terminationGracePeriodSeconds
          value: 20
  - it: should override priorityClassName
    set:
      priorityClassName: high-priority
    asserts:
      - equal:
          path: spec.template.spec.priorityClassName
          value: high-priority
  - it: should override topologySpreadConstraints
    set:
      topologySpreadConstraints:
        - maxSkew: 1
          topologyKey: kubernetes.io/hostname
          whenUnsatisfiable: DoNotSchedule
          labelSelector:
            matchLabels:
              control-plane: controller-manager
              app.kubernetes.io/name: powerdns-operator
    asserts:
      - equal:
          path: spec.template.spec.topologySpreadConstraints
          value:
            - maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: DoNotSchedule
              labelSelector:
                matchLabels:
                  control-plane: controller-manager
                  app.kubernetes.io/name: powerdns-operator
  - it: should override tolerations
    set:
      tolerations:
        - key: node-role.kubernetes.io/master
          operator: Exists
          effect: NoSchedule
    asserts:
      - equal:
          path: spec.template.spec.tolerations
          value:
            - key: node-role.kubernetes.io/master
              operator: Exists
              effect: NoSchedule
  - it: should override affinity
    set:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: kubernetes.io/e2e-az-name
                    operator: In
                    values:
                      - e2e-az1
                      - e2e-az2
    asserts:
      - equal:
          path: spec.template.spec.affinity
          value:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                  - matchExpressions:
                      - key: kubernetes.io/e2e-az-name
                        operator: In
                        values:
                          - e2e-az1
                          - e2e-az2
  - it: should override nodeSelector
    set:
      nodeSelector:
        disktype: ssd
    asserts:
      - equal:
          path: spec.template.spec.nodeSelector
          value:
            disktype: ssd
  - it: should override dnsPolicy
    set:
      dnsPolicy: ClusterFirstWithHostNet
    asserts:
      - equal:
          path: spec.template.spec.dnsPolicy
          value: ClusterFirstWithHostNet
  - it: should override dnsConfig
    set:
      dnsConfig:
        nameservers:
          - 1.1.1.1
          - 8.8.8.8
    asserts:
      - equal:
          path: spec.template.spec.dnsConfig
          value:
            nameservers:
              - 1.1.1.1
              - 8.8.8.8
