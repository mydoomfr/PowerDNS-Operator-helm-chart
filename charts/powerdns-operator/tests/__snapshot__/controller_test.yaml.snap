should match snapshot of default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: powerdns-operator
        app.kubernetes.io/version: 0.5.0
        helm.sh/chart: powerdns-operator-0.5.1
      name: RELEASE-NAME-powerdns-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: powerdns-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/component: controller
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: powerdns-operator
            app.kubernetes.io/version: 0.5.0
            helm.sh/chart: powerdns-operator-0.5.1
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8080
                - --pdns-api-url=https://powerdns.example.local:8081
                - --pdns-api-vhost=localhost
                - --pdns-api-timeout=10
              command:
                - /manager
              envFrom:
                - secretRef:
                    name: powerdns-api-credentials
              image: ghcr.io/powerdns-operator/powerdns-operator:v0.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: powerdns-operator
              ports:
                - containerPort: 8080
                  name: http-metrics
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
          dnsPolicy: ClusterFirst
          hostNetwork: false
          securityContext:
            runAsNonRoot: true
          serviceAccountName: RELEASE-NAME-powerdns-operator
          terminationGracePeriodSeconds: 10
should match snapshot with CA certificate configuration:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: powerdns-operator
        app.kubernetes.io/version: 0.5.0
        helm.sh/chart: powerdns-operator-0.5.1
      name: RELEASE-NAME-powerdns-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: powerdns-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/component: controller
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: powerdns-operator
            app.kubernetes.io/version: 0.5.0
            helm.sh/chart: powerdns-operator-0.5.1
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8080
                - --pdns-api-url=https://powerdns.example.local:8081
                - --pdns-api-vhost=localhost
                - --pdns-api-timeout=10
                - --pdns-api-ca-path=/certs/ca.crt
              command:
                - /manager
              envFrom:
                - secretRef:
                    name: powerdns-api-credentials
              image: ghcr.io/powerdns-operator/powerdns-operator:v0.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: powerdns-operator
              ports:
                - containerPort: 8080
                  name: http-metrics
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /certs
                  name: ca-cert
                  readOnly: true
          dnsPolicy: ClusterFirst
          hostNetwork: false
          securityContext:
            runAsNonRoot: true
          serviceAccountName: RELEASE-NAME-powerdns-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - configMap:
                items:
                  - key: ca.crt
                    path: ca.crt
                name: RELEASE-NAME-powerdns-operator-ca-cert
              name: ca-cert
should match snapshot with existing ConfigMap configuration:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: powerdns-operator
        app.kubernetes.io/version: 0.5.0
        helm.sh/chart: powerdns-operator-0.5.1
      name: RELEASE-NAME-powerdns-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: powerdns-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/component: controller
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: powerdns-operator
            app.kubernetes.io/version: 0.5.0
            helm.sh/chart: powerdns-operator-0.5.1
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8080
                - --pdns-api-url=https://powerdns.example.local:8081
                - --pdns-api-vhost=localhost
                - --pdns-api-timeout=10
                - --pdns-api-ca-path=/certs/ca.crt
              command:
                - /manager
              envFrom:
                - secretRef:
                    name: powerdns-api-credentials
              image: ghcr.io/powerdns-operator/powerdns-operator:v0.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: powerdns-operator
              ports:
                - containerPort: 8080
                  name: http-metrics
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /certs
                  name: ca-cert
                  readOnly: true
          dnsPolicy: ClusterFirst
          hostNetwork: false
          securityContext:
            runAsNonRoot: true
          serviceAccountName: RELEASE-NAME-powerdns-operator
          terminationGracePeriodSeconds: 10
          volumes:
            - configMap:
                items:
                  - key: ca.crt
                    path: ca.crt
                name: my-ca-configmap
              name: ca-cert
should match snapshot with insecure TLS configuration:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: powerdns-operator
        app.kubernetes.io/version: 0.5.0
        helm.sh/chart: powerdns-operator-0.5.1
      name: RELEASE-NAME-powerdns-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: powerdns-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/component: controller
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: powerdns-operator
            app.kubernetes.io/version: 0.5.0
            helm.sh/chart: powerdns-operator-0.5.1
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8080
                - --pdns-api-url=https://powerdns.example.local:8081
                - --pdns-api-vhost=localhost
                - --pdns-api-timeout=10
                - --pdns-api-insecure
              command:
                - /manager
              envFrom:
                - secretRef:
                    name: powerdns-api-credentials
              image: ghcr.io/powerdns-operator/powerdns-operator:v0.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: powerdns-operator
              ports:
                - containerPort: 8080
                  name: http-metrics
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
          dnsPolicy: ClusterFirst
          hostNetwork: false
          securityContext:
            runAsNonRoot: true
          serviceAccountName: RELEASE-NAME-powerdns-operator
          terminationGracePeriodSeconds: 10
